File: bitmarket-server/src/main/java/com/bitwise/bitmarket/registry/CommandLine.java
package com.bitwise.bitmarket.registry;

import com.beust.jcommander.JCommander;
import com.beust.jcommander.Parameter;

public class CommandLine {

    public static final int DEFAULT_PORT = 8080;

    @Parameter(
            names = { "-p", "--port" },
            description = "Port to use. Default value: " + DEFAULT_PORT
    )
    public int port = DEFAULT_PORT;

    public static CommandLine fromArgList(String[] args) {
        CommandLine cli = new CommandLine();
        new JCommander(cli, args);
        return cli;
    }
}

File: bitmarket-server/src/main/scala/com/bitwise/bitmarket/CommandLine.scala
package com.bitwise.bitmarket

import com.beust.jcommander.{JCommander, Parameter}

class CommandLine {
  @Parameter(names = Array("-p", "--port"))
  var port: Int = CommandLine.DefaultPort
}

object CommandLine {

  val DefaultPort = 8080

  def fromArgList(args: Seq[String]): CommandLine = {
    val cli = new CommandLine
    new JCommander(cli, args: _*)
    cli
  }
}
